examples/qaoa.py

```diff
@@ -151,8 +150,12 @@ def qaoa_max_cut_circuit(qubits, betas, gammas,
         cirq.measure(*qubits, key='m'))
 
 
-def cut_value(bitstring, graph):
-    return sum(bitstring[i] != bitstring[j] for i, j in graph.edges)
+def cut_values(bitstrings, graph):
+    mat = networkx.adjacency_matrix(graph, nodelist=sorted(graph.nodes))
+    vecs = (-1)**bitstrings
+    vals = 0.5 * np.einsum('...i,...i', vecs, (mat * vecs.T).T)
```

Comment:
do you need einsum here? I think `'...i,...i'` is the default numpy broadcasting behavior (?) Could you achieve the same effect with

```python
vecs * (mat * vecs.T).T)
```

**or**

can you write a better einsum to avoid one or both of the `.T`'s

Reply:
or something like 
```python
vecs.T @ mat @ vecs
```